### ISSUE TITLE
[a11y]: `SidePanel` stories action button text should be changed

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-products#5772]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

`2.45.0-rc.1`

### Automated testing tool and ruleset

Manual

### Assistive technology

JAWS

### Description

The action button in the `SidePanel` stories are getting flagged because the screenreader announces the button text followed by the role ("button"). Currently the button text is "Primary button", which causes the the screenreader to announce "Primary button button".

We should just change the primary action to "Submit" and the secondary action to "Cancel".

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic

### Steps to reproduce

Visible in storybook 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
